tests/publiccloud/ipa.pm

```diff
@@ -104,28 +129,29 @@ sub run {
           . "--region 'eu-central-1' "
           . "-u ec2-user "
           . "--ssh-private-key QA_SSH_KEY.pem "
-          . "--ssh-key-name QA_SSH_KEY "
+          . "--ssh-key-name '" . $ssh_key_name . "' "
           . "--results-dir ipa_results "
           . "test_sles "
           . "test_sles_ec2 "
           . "test_sles_on_demand ",
         timeout => 600
     );
 
+    assert_script_run("aws ec2 delete-key-pair --key-name " . $ssh_key_name);
     save_logs;
 }
 
 sub post_fail_hook {
+    my ($self) = @_;
 
     save_logs;
 
-    # Terminate all instances
-    my $out = script_output("aws ec2 describe-instances");
-    for my $line (split(/\r?\n/, $out)) {
-        if ($line =~ /"InstanceId":\s+"([^"]+)"/) {
-            script_run("aws ec2 terminate-instances --instance-ids $1");
-        }
+    # Terminate instance
+    if (defined($self->{'ipa_instance_id'})) {
+        script_run("aws ec2 terminate-instances --instance-ids " . $self->{'ipa_instance_id'});
```

Comment:
Why would you like to ignore failures in this call? If the termination fails, isn't that something critical we want to catch? Or could it be that the instance already terminated in a different way in before? I would be more explicit about this and in general recommend to use `script_run` only when you use the return code

Reply:
right, I will change that.